src/units/key_manager.rs

```diff
@@ -295,24 +197,151 @@ impl KeyManager {
+        // changes `dnst keyset create` made on disk? Will leaving them behind
+        // mean that a subsequent attempt to again create the zone after
+        // resolving whatever failure occurred below will then fail because
+        // the `dnst keyset create`d state already exists?
+
+        if let Some(kmip_server_id) = kmip_server_id {
+            let kmip_server_state_path = kmip_server_state_dir.join(kmip_server_id);
+
+            log::debug!("Reading KMIP server state from '{kmip_server_state_path}'");
+            let f = File::open(&kmip_server_state_path).unwrap();
+            let kmip_server: KmipServerState = serde_json::from_reader(f).map_err(|err| {
+                ZoneAddError::Other(format!(
+                    "Unable to read KMIP server state from file '{kmip_server_state_path}': {err}"
+                ))
+            })?;
+
+            let KmipServerState {
+                server_id,
+                ip_host_or_fqdn,
+                port,
+                insecure,
+                connect_timeout,
+                read_timeout,
+                write_timeout,
+                max_response_bytes,
+                key_label_prefix,
+                key_label_max_bytes,
+                has_credentials,
+            } = kmip_server;
+
+            let mut cmd = self.keyset_cmd(name.clone());
+
+            // TODO: This command should get issued _after_ keyset create
+            // but _before_ keyset init, both of which are issued above.
+            cmd.arg("kmip")
+                .arg("add-server")
+                .arg(server_id.clone())
+                .arg(ip_host_or_fqdn)
+                .arg("--port")
+                .arg(port.to_string())
+                .arg("--connect-timeout")
+                .arg(format!("{}s", connect_timeout.as_secs()))
+                .arg("--read-timeout")
+                .arg(format!("{}s", read_timeout.as_secs()))
+                .arg("--write-timeout")
+                .arg(format!("{}s", write_timeout.as_secs()))
+                .arg("--max-response-bytes")
+                .arg(max_response_bytes.to_string())
+                .arg("--key-label-max-bytes")
+                .arg(key_label_max_bytes.to_string());
+
+            if insecure {
+                cmd.arg("--insecure");
             }
 
-            _ => Ok(()), // not for us
+            if has_credentials {
+                cmd.arg("--credential-store")
+                    .arg(kmip_credentials_store_path.as_str());
+            }
+
+            if let Some(key_label_prefix) = key_label_prefix {
+                cmd.arg("--key-label-prefix").arg(key_label_prefix);
+            }
+
+            // TODO: --client-cert, --client-key, --server-cert and --ca-cert
+            cmd.output()
+                .map_err(|err| ZoneAddError::Other(err.reason))?;
         }
+
+        // Set config
+        let config_commands = policy_to_commands(&policy);
```

Comment:
This will conflict with the key imports. Let me know when I need to update that or if you want to merge key imports first.